File: android/app/src/main/java/com/aitutor/app/domain/model/ExamMode.kt
package com.aitutor.app.domain.model

import androidx.compose.ui.graphics.Color
import com.aitutor.app.ui.theme.IeltsBlue
import com.aitutor.app.ui.theme.ToeflRed

/**
 * Which exam the user is currently preparing for.
 *
 * Drives content selection (topic banks, scoring rubrics, task types) across all
 * skill modules. Persisted in [com.aitutor.app.data.local.UserPreferences].
 */
enum class ExamMode(
    val displayName: String,
    val tagline: String,
    val brandColor: Color,
    val scoreLabel: String,
    val maxScore: Double
) {
    IELTS(
        displayName = "IELTS",
        tagline = "Academic & General • 0–9 band",
        brandColor = IeltsBlue,
        scoreLabel = "Band",
        maxScore = 9.0
    ),
    TOEFL(
        displayName = "TOEFL iBT",
        tagline = "Academic • 0–120 total",
        brandColor = ToeflRed,
        scoreLabel = "Score",
        maxScore = 120.0
    );

    companion object {
        fun fromName(name: String?): ExamMode? =
            entries.firstOrNull { it.name.equals(name, ignoreCase = true) }
    }
}

File: ios/AITutor/Domain/Model/ExamMode.swift
import SwiftUI

/// Which exam the user is currently preparing for. Drives content selection
/// (topic banks, scoring rubrics, task types) across all skill modules.
/// Persisted in `UserPreferences`.
enum ExamMode: String, CaseIterable, Identifiable, Codable {
    case ielts = "IELTS"
    case toefl = "TOEFL"

    var id: String { rawValue }

    var displayName: String {
        switch self {
        case .ielts: return "IELTS"
        case .toefl: return "TOEFL iBT"
        }
    }

    var tagline: String {
        switch self {
        case .ielts: return "Academic & General • 0–9 band"
        case .toefl: return "Academic • 0–120 total"
        }
    }

    var brandColor: Color {
        switch self {
        case .ielts: return Theme.Colors.ieltsBlue
        case .toefl: return Theme.Colors.toeflRed
        }
    }

    var scoreLabel: String {
        switch self {
        case .ielts: return "Band"
        case .toefl: return "Score"
        }
    }

    var maxScore: Double {
        switch self {
        case .ielts: return 9.0
        case .toefl: return 120.0
        }
    }

    static func fromName(_ name: String?) -> ExamMode? {
        guard let name else { return nil }
        return ExamMode(rawValue: name.uppercased())
    }
}
